src/os/bluestore/bluestore_common.h

```diff
@@ -24,7 +24,7 @@ void apply_for_bitset_range(uint64_t off,
   uint64_t granularity,
   Bitset &bitset,
   Func f) {
-  auto end = round_up_to(off + len, granularity) / granularity;
+  auto end = (off + len) / granularity;
```

Comment:
The issue is valid indeed but IMO this isn't a proper way of fixing it. apply_for_bitset_range() is a general purpose function and it can get unaligned chunks (offset~length) for processing and hence it must be able to operate on short block tails. Instead your current fix bypasses them.
So more correct fix would be to adjust owned_blocks/used blocks resize calls at  _replay beginning.

Reply:
are you mean owned_blocks[i].resize(round_up_to(bdev[i]->get_size(), alloc_size[i]) / alloc_size[i])?  But this can increase real space.

Reply:
Yeah, and the same for 'used_blocks'
These bitsets are used for allocation verification only and updated resize wouldn't impact anything but this verification.  From verification procedure point of view bluefs owns/uses some chunk even when the latter is owned/used just partially. So one needs to track the last chunk and hence have corresponding bit in the bitsets.